src/Api/PubnubApi/EventEngine/Subscribe/Effects/ReceivingEffectHandler.cs

```diff
@@ -14,69 +10,43 @@
 namespace PubnubApi.EventEngine.Subscribe.Effects
 {
     public class ReceivingEffectHandler:
-        EffectDoubleCancellableHandler<ReceiveMessagesInvocation, ReceiveReconnectInvocation, CancelReceiveMessagesInvocation, CancelReceiveReconnectInvocation>
+        EffectCancellableHandler<ReceiveMessagesInvocation, CancelReceiveMessagesInvocation>
     {
-        private PNConfiguration pubnubConfiguration;
         private SubscribeManager2 manager;
         private EventQueue eventQueue;
-        
-        private Delay retryDelay = new Delay(0);
 
-        internal ReceivingEffectHandler(PNConfiguration pubnubConfiguration, SubscribeManager2 manager, EventQueue eventQueue)
+        internal ReceivingEffectHandler(SubscribeManager2 manager, EventQueue eventQueue)
         {
-            this.pubnubConfiguration = pubnubConfiguration;
             this.manager = manager;
             this.eventQueue = eventQueue;
         }
 
-        public override Task Run(ReceiveReconnectInvocation invocation)
-        {
-			var retryConfiguration = pubnubConfiguration.RetryConfiguration;
-			if (retryConfiguration == null)
-            {
-                eventQueue.Enqueue(new ReceiveReconnectGiveUpEvent() { Status = new PNStatus(PNStatusCategory.PNCancelledCategory) });
-            }
-			else if (!retryConfiguration.RetryPolicy.ShouldRetry(invocation.AttemptedRetries, invocation.Reason))
-            {
-				eventQueue.Enqueue(new ReceiveReconnectGiveUpEvent() { Status = new PNStatus(PNStatusCategory.PNCancelledCategory) });
-			}
-            else
-            {
-                retryDelay = new Delay(retryConfiguration.RetryPolicy.GetDelay(invocation.AttemptedRetries, invocation.Reason, null));
-                // Run in the background
-                retryDelay.Start().ContinueWith((_) => this.Run((ReceiveMessagesInvocation)invocation));
-            }
-
-            return Utils.EmptyTask;
-        }
-
-        public override bool IsBackground(ReceiveReconnectInvocation invocation)
-        {
-            return true;
-        }
-
         public override async Task Run(ReceiveMessagesInvocation invocation)
         {
             var response = await MakeReceiveMessagesRequest(invocation);
             var cursor = new SubscriptionCursor()
             {
-                Region = response.Item1?.Timetoken.Region,
-                Timetoken = response.Item1?.Timetoken.Timestamp
+                Region = response.Item1?.Timetoken?.Region,
+                Timetoken = response.Item1?.Timetoken?.Timestamp
```

Comment:
Maybe `null` check should be done at the beggining. 
I know that it won't faile because of the `?` mark but still, there is no need to setup new variables if the `response.Item2 is null` return the whole function.

Reply:
Yes, I think null conditional operator here is not necessary and check before creating cursor object makes sense.

Reply:
Updated!!!